### PR TITLE
PAAS-3303 allow configuring kubernetes rollout timeout per project

### PR DESCRIPTION
**Pull request by grosser**
we have a few projects that take forever, so allow them to still deploy

@zendesk/compute @zendesk/bre @craig-day 

Screen Shot 2019-05-08 at 4 37 03 PM

### CODE REVIEW

**Review by uthark — APPROVED** [review]

File: plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<fieldset>
```
Comment (adammw) [inline plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb]: so this is per project, not per stage?

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@ def execute(*)
       end
 
       if deploys.any?
-        return false unless deploy_and_watch(deploys, timeout: WAIT_FOR_LIVE)
+        return false unless deploy_and_watch(deploys, timeout: @job.project.kubernetes_rollout_timeout || WAIT_FOR_LIVE)
```
Comment (adammw) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: rename constant DEFAULT_WAIT_FOR_LIVE?

Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: or default wait timeout

**Review by adammw — APPROVED** [review]

File: plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb (hunk 1 above)
Comment (grosser) [reply plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb]: usually all stages deploy the same slow software, so would make sense to have it on the project 🤷‍♂

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb (hunk 2 above)
Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: -> DEFAULT_ROLLOUT_TIMEOUT

File: plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb (hunk 1 above)
Comment (adammw) [reply plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb]: I'm thinking more if say production or one pod has way more replicas than others, but yes i guess that's a different problem for us to solve a different way via "making progress" events or the like...

Comment (grosser) [reply plugins/kubernetes/app/views/samson_kubernetes/_project_form.html.erb]: can always crank this high as an escape hatch then :D

**Pull request merged**